core/src/main/scala/kafka/log/Log.scala

```diff
@@ -181,25 +191,38 @@ class Log(val dir: File,
         // if its a log file, load the corresponding log segment
         val start = filename.substring(0, filename.length - LogFileSuffix.length).toLong
         val indexFile = Log.indexFilename(dir, start)
+        val timeIndexFile = Log.timeIndexFilename(dir, start)
         val segment = new LogSegment(dir = dir,
                                      startOffset = start,
                                      indexIntervalBytes = config.indexInterval,
                                      maxIndexSize = config.maxIndexSize,
                                      rollJitterMs = config.randomSegmentJitter,
                                      time = time,
                                      fileAlreadyExists = true)
+        if (indexFile.exists() && timeIndexFile.exists()) {
 
-        if(indexFile.exists()) {
           try {
-              segment.index.sanityCheck()
+            segment.index.sanityCheck()
           } catch {
             case e: java.lang.IllegalArgumentException =>
-              warn("Found a corrupted index file, %s, deleting and rebuilding index...".format(indexFile.getAbsolutePath))
+              warn(s"Found a corrupted index file, deleting ${timeIndexFile.getAbsolutePath}, " +
+                s"${indexFile.getAbsolutePath} and rebuilding index...")
               indexFile.delete()
+              timeIndexFile.delete()
               segment.recover(config.maxMessageSize)
           }
-        }
-        else {
+
+          try {
+            segment.timeIndex.sanityCheck()
+          } catch {
+            case e: java.lang.IllegalArgumentException =>
+              warn(s"Found a corrupted time index file, deleting ${timeIndexFile.getAbsolutePath}, " +
+                  s"${indexFile.getAbsolutePath} and rebuilding index...")
+              indexFile.delete()
+              timeIndexFile.delete()
+              segment.recover(config.maxMessageSize)
+          }
+        } else {
```

Comment:
Interesting. Then, in the code, perhaps we should just get rid of the check of the existence of the offset and the time index files since they are guaranteed to be present after LogSegment is created? I agree the case that people manually delete the index files is rare.